dev/tests/agent.rs

```diff
@@ -253,6 +256,66 @@ async fn drone_sends_draining_status() {
         .unwrap();
 }
 
+#[allow(dead_code)]
+struct Ctx {
+    nats_server: Nats,
+    nats_connection: TypedNats,
+    controller: MockController,
+    agent: Agent,
+}
+
+async fn do_spawn_request(mut request: SpawnRequest) -> (Ctx, BackendStateSubscription) {
```

Comment:
Since `Ctx` exists to be returned by this function, how about putting the state subscription as a field on `Ctx` rather than returning a tuple?

Reply:
I was thinking there'd be other things that also eventually returned TestFixture, and also their individual things in a tuple.